File: webstore/webstore-java/src/main/java/com/renan/webstore/business/MailService.java
package com.renan.webstore.business;

import java.util.List;

public interface MailService {
    
    void sendEmail(String destination, String subject, String message) throws BusinessException;
    
    void sendEmail(List<String> destinationList, String subject, String message) throws BusinessException;
    
    void sendNewPasswordEmail(String destination, String newPasswordCode) throws BusinessException;
    
    void sendActivationEmail(String destination, String activationCode) throws BusinessException;
    
}

File: webstore/webstore-kotlin/src/main/kotlin/com/renan/webstore/business/MailService.kt
package com.renan.webstore.business

interface MailService {

    @Throws(BusinessException::class)
    fun sendEmail(destination: String, subject: String, message: String)

    @Throws(BusinessException::class)
    fun sendEmail(destinationList: List<String>, subject: String, message: String)

    @Throws(BusinessException::class)
    fun sendNewPasswordEmail(destination: String, newPasswordCode: String)

    @Throws(BusinessException::class)
    fun sendActivationEmail(destination: String, activationCode: String)

}